lib/auth/oidc/client_assertion.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package oidc
+
+import (
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"os"
+
+	gojwt "github.com/golang-jwt/jwt/v5"
+	cass "github.com/hashicorp/cap/oidc/clientassertion"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+type ActiveKeyGetter interface {
+	GetActiveKey() (key *rsa.PrivateKey, id string, err error)
+}
```

Comment:
I see why you've done this, but this exposes the encrypter API to the rest of the code base, which potentially may be waiting on I/O due to keyring replication. It might be a good idea to have the RPC handler (which already has access to the keyring) simply get the key and pass it as a parameter to `BuildClientAssertionJWT`. As far as I can tell there's only one caller anyways.